### PR TITLE
meta(codeowners): Add code owners for `.github`

### PR DESCRIPTION
**Pull request by billyvg**

### CODE REVIEW

**Review by BYK — CHANGES_REQUESTED** [review]
@getsentry/releases should own some of the workflows. Why do we need CODEOWNERS for this. Did someone break something?

**Comment by billyvg** [comment]
> @getsentry/releases should own some of the workflows. Why do we need CODEOWNERS for this. Did someone break something?

Why not have code owners for it? Feel free to add `@getsentry/releases` to workflows that you see fit. It's already an owner of the release workflow: https://github.com/getsentry/sentry/pull/22234/files#diff-3d36a1bf06148bc6ba1ce2ed3d19de32ea708d955fed212c0d27c536f0bd4da7L34

**Review by BYK — APPROVED** [review]
Ooops, sorry I forgot about that.

As long as we don't block on reviews from codeowners, this is actually fine to get some default reviewers, yeah.

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -35,7 +35,7 @@
 /docker                         @getsentry/releases
 setup.py                        @getsentry/releases
 setup.cfg                       @getsentry/releases
-# TODO: owners-build (for ci)
+/.github/           @getsentry/owners-sentry-dev
```
Comment (joshuarli) [inline .github/CODEOWNERS:38]: Move this down to the `# Dev` section.

**Review by joshuarli — APPROVED** [review]

**Pull request merged**